Packages/Tracking/Core/Runtime/Scripts/Utils/AndroidServiceBinder.cs

```diff
@@ -0,0 +1,94 @@
+namespace Leap.Unity
+{
+    using System;
+    using UnityEngine;
+
+    public static class AndroidServiceBinder
+    {
+#if UNITY_ANDROID
+
+        private static bool isBound = false;
+        public static bool IsBound
+        { 
+            get { return isBound; }
+            private set { isBound = value; }
+        }
+        static AndroidJavaObject _serviceBinder;
+        static AndroidJavaClass unityPlayer;
+        static AndroidJavaObject activity;
+        static AndroidJavaObject context;
+        static ServiceCallbacks serviceCallbacks;
+
+        public static bool Bind()
+        {
+            bool isBound = _serviceBinder?.Call<bool>("isBound") ?? false;
+
+            if (!isBound)
+            {
+                isBound = TryBind();
```

Comment:
This doesn't look like it handles the failure of `TryBind()`. In this situation, it would set the Application.quitting event even though binding was unsuccessful. this would then call `Unbind()` on application termination even though we never successfully bound.